### PR TITLE
[MERGE] base: add lastcall to cron and pass it to jobs

### PR DESCRIPTION
**Pull request by xmo-odoo**
Many crons process a dedicated queue of some sort (and either delete or
disable the items once processed) but some may not want or be able to do
that. In that case maintaining some sort of time range is convenient,
and the cron itself doing so avoids weird solutions (like global
settings).

Task 1912648

### CODE REVIEW

File: addons/calendar/models/calendar.py [hunk 1]
```diff
@@ -353,8 +353,13 @@ def do_check_alarm_for_one_date(self, one_date, event, event_maxdelta, in_the_ne
 
     @api.model
     def get_next_mail(self):
-        now = fields.Datetime.to_string(fields.Datetime.now())
-        last_notif_mail = self.env['ir.config_parameter'].sudo().get_param('calendar.last_notif_mail', default=now)
+        if self.env.context.get('lastcall'):
+            last_notif_mail = fields.Datetime.to_string(self.env.context['lastcall'])
+        else:
+            last_notif_mail = self.env['ir.config_parameter'].sudo().get_param(
```
Comment (rco-odoo) [inline addons/calendar/models/calendar.py]: Do you think this backward-compatibility is worth it?
I would remove that part and use `self.env.context.get('lastcall') or fields.Datetime.now()` in the expression above.
We can leave a migration note that says: set the `lastcall` of the `ir.cron` to the value of the `ir.config_parameter`.

Comment (xmo-odoo) [reply addons/calendar/models/calendar.py]: @rco-odoo removed the BC bit

**Comment by xmo-odoo** [comment]
robodoo r+

**Comment by xmo-odoo** [comment]
robodoo rebase-ff

**Pull request closed without merging**